packages/reactivity/src/watch.ts

```diff
@@ -215,8 +215,10 @@ export function watch(
     effect.stop()
     if (scope) {
       remove(scope.effects, effect)
+      remove(scope.cleanups, watchHandle)
     }
   }
+  onScopeDispose(watchHandle, true)
```

Comment:
I'm wondering whether there's an alternative to this. Registering the watch handle feels a bit heavy.

Could we achieve the same result by just emptying the `effects` array inside `stop`? e.g. `this.effects.length = 0`? Maybe `cleanups` too?

Reply:
Yes, I have tried cleaning `effects` and `cleanups` to achieve the same results

```
this.effects.length = 0
this.cleanups.length = 0
```

Reply:
Done